dist/images/kubectl-ko

```diff
@@ -949,159 +941,107 @@ env-check(){
   done
 }
 
 log_ovn_ovs(){
   component_param=$1
-  sub_component_param=$2
-  tail_lines_param=$3
+  echo "logging $component_param log"
   podNames=`kubectl get pod -n kube-system -l app=ovs -o 'jsonpath={.items[*].metadata.name}'`
   for pod in $podNames; do
-    echo "************************************************"
-    echo " component $component_param $sub_component_param.log pod $pod log"
-    echo "************************************************"
-
-    if [[ "$tail_lines_param" == "all" ]]; then
-      kubectl exec $pod -n kube-system -- cat /var/log/$component_param/$sub_component_param.log 2>&1 || :
-    else
-      kubectl exec $pod -n kube-system -- tail -n $tail_lines_param /var/log/$component_param/$sub_component_param.log 2>&1 || :
-    fi
+    nodeName=$(kubectl get pod "$pod" -n kube-system -o jsonpath={.spec.nodeName})
+    mkdir -p ./kubectl-ko-log/$nodeName/$component_param
+    kubectl cp $pod:/var/log/$component_param ./kubectl-ko-log/$nodeName/$component_param -n kube-system > /dev/null || :
   done
 }
 
 log_linux(){
-  sub_component_param=$1
-  tail_lines_param=$2
+  component_param=$1
+  sub_component_param=$2
+  echo "logging $component_param $sub_component_param log"
   podNames=`kubectl get pod -n kube-system -l app=ovs -o 'jsonpath={.items[*].metadata.name}'`
   for pod in $podNames; do
-    echo "************************************************"
-    echo " component $component sub_component $sub_component_param pod $pod log"
-    echo "************************************************"
+    nodeName=$(kubectl get pod "$pod" -n kube-system -o jsonpath={.spec.nodeName})
+    mkdir -p ./kubectl-ko-log/$nodeName/$component_param
     if [[ "$sub_component_param" == "dmesg" ]]; then
-      kubectl exec $pod -n kube-system -- dmesg -T $( [ "$tail_lines_param" == "all" ] || echo "| tail -n $tail_lines_param" ) || :
+      kubectl exec $pod -n kube-system -- dmesg -T > ./kubectl-ko-log/$nodeName/$component_param/$sub_component_param || :
     elif [[ "$sub_component_param" == "iptables" ]]; then
-      echo "******************legacy filter ************************"
-      kubectl exec $pod -n kube-system -- iptables -S || :
-      echo "*********************nft filter ************************"
-      kubectl exec $pod -n kube-system -- /usr/sbin/iptables-nft -S || :
+      echo "******************legacy filter ************************" > ./kubectl-ko-log/$nodeName/$component_param/$sub_component_param
+      kubectl exec $pod -n kube-system -- /usr/sbin/iptables-legacy -S >> ./kubectl-ko-log/$nodeName/$component_param/$sub_component_param || :
+      echo "*********************nft filter ************************" >> ./kubectl-ko-log/$nodeName/$component_param/$sub_component_param
+      kubectl exec $pod -n kube-system -- /usr/sbin/iptables-nft -S >> ./kubectl-ko-log/$nodeName/$component_param/$sub_component_param 2>/dev/null || :
```

Comment:
I suggest splitting iptables rules by iptables mode: save legacy rules to iptables-legacy.txt and nft rules to nft-legacy.txt

Reply:
add suffix with .log